buildchain/buildchain/packaging.py

```diff
@@ -181,269 +288,381 @@ def clean() -> None:
+    for repository in DEB_REPOSITORIES[releasever]:
         if next(repository.pkgdir.glob('*.deb'), False):
             yield from repository.execution_plan
 
+@doit.create_after(executed='_download_ubuntu_18_04_packages')  # type: ignore
+def task__build_ubuntu_18_04_repositories() -> Iterator[types.TaskDict]:
+    """Build Ubuntu 18.04 repositories."""
+    return _build_deb_repositories('18.04')
+
 # }}}
 # RPM packages and repository {{{
 
 # Packages to build, per repository.
-def _rpm_package(name: str, sources: List[Path]) -> targets.RPMPackage:
+def _rpm_package(
+    name: str, releasever: str, sources: List[Path]
+) -> targets.RPMPackage:
     try:
-        pkg_info = versions.RPM_PACKAGES_MAP[name]
+        pkg_info = versions.REDHAT_PACKAGES_MAP[releasever][name]
     except KeyError as exc:
         raise ValueError(
-            'Missing version for package "{}"'.format(name)
+            'Missing version for package "{}" for release "{}"'.format(
+                name, releasever
+            )
         ) from exc
 
     # In case the `release` is of form "{build_id}.{os}", which is standard
     build_id_str, _, _ = pkg_info.release.partition('.')
 
     return targets.RPMPackage(
-        basename='_build_rpm_packages',
+        basename='_build_redhat_{0}_packages'.format(releasever),
         name=name,
         version=pkg_info.version,
         build_id=int(build_id_str),
         sources=sources,
-        builder=builder.RPM_BUILDER,
+        builder=builder.RPM_BUILDER[releasever],
+        releasever=releasever,
         task_dep=[
-            '_package_mkdir_rpm_root',
-            '_build_builder:{}'.format(builder.RPM_BUILDER.name)
+            '_package_mkdir_redhat_{0}_root'.format(releasever),
+            '_build_builder:{}'.format(builder.RPM_BUILDER[releasever].name)
         ],
     )
 
-
 def _rpm_repository(
-    name: str, packages: Optional[Sequence[targets.RPMPackage]]=None
+    name: str,
+    releasever: str,
+    packages: Optional[Sequence[targets.RPMPackage]]=None
 ) -> targets.RPMRepository:
     """Return a RPM repository object.
 
     Arguments:
         name:     repository name
         packages: list of locally built packages
     """
-    mkdir_task = '_package_mkdir_rpm_iso_root'
-    download_task = '_download_rpm_packages'
+    mkdir_task = '_package_mkdir_redhat_{0}_iso_root'.format(releasever)
+    download_task = '_download_redhat_{0}_packages'.format(releasever)
     return targets.RPMRepository(
-        basename='_build_rpm_repositories',
+        basename='_build_redhat_{0}_repositories'.format(releasever),
         name=name,
-        builder=builder.RPM_BUILDER,
+        releasever=releasever,
+        builder=builder.RPM_BUILDER[releasever],
         packages=packages,
         task_dep=[download_task if packages is None else mkdir_task],
     )
 
 
 # Calico Container Network Interface Plugin.
-CALICO_RPM = _rpm_package(
+REDHAT_7_CALICO_RPM = _rpm_package(
```

Comment:
There's a difference between how we handle the `7` and `8` package for Calico. Would it make sense to turn this into a function taking a `releasever`, then call it twice in `RPM_TO_BUILD` construction?

Reply:
Totally